src/control/controlobjectscript.cpp

```diff
@@ -130,10 +130,15 @@ void ControlObjectScript::disconnectAllConnectionsToFunction(const QJSValue& fun
 void ControlObjectScript::slotValueChanged(double value, QObject*) {
     // Make a local copy of m_connectedScriptFunctions first.
     // This allows a script to disconnect a callback from inside the
-    // the callback. Otherwise the this may crash since the disconnect call
-    // happens during conn.function.call() in the middle of the loop below.
-    const QVector<ScriptConnection> connections = m_scriptConnections;
+     const QVector<ScriptConnection> connections = m_scriptConnections;
+    // Flag to inhibit triggering connections during callback execution
+     bool callbackExecuting = false;
```

Comment:
callbackExecuting is a local variable, which is notvisible outside of this function. Therefore it can't work.

Reply:
what if I specify this variable as a data member of class ControlObjectScript

Reply:
This could work! 
In general, you need to create a test case, which conntains a trigger() call of itself in it's callback. Without, you can't test if your code does, what it should.

Reply:
You still have two independent variables, it can only work if you write/read into the same membe variable

Reply:
can i make the ControllerScriptInterfaceLegacy as a friend class of ControlObjectScript then i will be able to create a single private data member inside ControlObjectScript.

Reply:
or i could make the m_callbackExecuting a public data member